### PR TITLE
Fix: do not break when pod status is empty (bsc#1161903)

### PR DESCRIPTION
**Pull request by mbologna**
## What does this PR change?

Fix a corner case in which the pod status are empty

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: internal

- [X] **DONE**

## Test coverage
- No tests: Salt runner

- [X] **DONE**

## Links

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests" (Test skipped, there are no changes to test)
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py [hunk 1]
```diff
@@ -44,13 +47,16 @@ def get_all_containers(kubeconfig=None, context=None):
...
+            for container in pod.status.container_statuses:
+                res_cont = dict()
+                res_cont['container_id'] = container.container_id
+                res_cont['image'] = container.image
+                res_cont['image_id'] = container.image_id
+                res_cont['pod_name'] = pod.metadata.name
+                res_cont['pod_namespace'] = pod.metadata.namespace
+                output['containers'].append(res_cont)
+        except:
+            log.error("Failed to parse pod container statuses")
```
Comment (mateiw) [inline susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py:60]: Could you also log the exception?

**Review by mateiw — CHANGES_REQUESTED** [review]

File: susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py [hunk 2]
```diff
@@ -44,13 +47,16 @@ def get_all_containers(kubeconfig=None, context=None):
...
-        for container in pod.status.container_statuses:
-            res_cont = dict()
-            res_cont['container_id'] = container.container_id
-            res_cont['image'] = container.image
-            res_cont['image_id'] = container.image_id
-            res_cont['pod_name'] = pod.metadata.name
-            res_cont['pod_namespace'] = pod.metadata.namespace
-            output['containers'].append(res_cont)
+        try:
+            for container in pod.status.container_statuses:
```
Comment (mateiw) [inline susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py:51]: IIUC `container_statuses` is empty? In this case maybe in addition to the `try...except` block, a check could be added to avoid the exception altogether.

Comment (mbologna) [reply susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py:51]: Yes, you are right

**Review by mateiw — APPROVED** [review]
LGTM :+1:
